app/resolver.go

```diff
@@ -15,9 +15,11 @@ var (
 	lookupIP = net.LookupIP
 )
 
-// Resolver periodically tries to resolve the IP addresses for a given
-// set of hostnames.
-type Resolver struct {
+type resolver interface {
```

Comment:
I'm all for abstraction and consistency, but I don't think this interface adds anything.  Its not as if the abstraction is used elsewhere to simplify something.  Rules like "constructors return interfaces" seem too general and remind me of all the horrors of Java.